test/extended/util/framework.go

```diff
@@ -189,7 +189,9 @@ var CheckImageStreamTagNotFoundFn = func(i *imageapi.ImageStream) bool {
 func WaitForADeployment(client kclient.ReplicationControllerInterface,
 	name string,
 	isOK, isFailed func(*kapi.ReplicationController) bool) error {
-	for {
+	startTime := time.Now()
+	endTime := startTime.Add(3 * time.Minute)
```

Comment:
I'm guessing the watch doesn't get closed because there's not much etcd
change log activity, so you don't come back to the top of the loop, so the
3 minute limit doesn't come into play in your particular scenario.

Ben Parees | OpenShift
On Dec 16, 2015 10:20 PM, "Gabe Montero" <notifications@github.com> wrote:

> In test/extended/util/framework.go
> <https://github.com/openshift/origin/pull/6329#discussion_r47865373>:
>
> > @@ -189,7 +189,9 @@ var CheckImageStreamTagNotFoundFn = func(i *imageapi.ImageStream) bool {
> >  func WaitForADeployment(client kclient.ReplicationControllerInterface,
> >  	name string,
> >  	isOK, isFailed func(*kapi.ReplicationController) bool) error {
> > -	for {
> > +	startTime := time.Now()
> > +	endTime := startTime.Add(3 * time.Minute)
>
> Will do - it worked for me locally but i did not measure how long it
> typically took
> … <#151adf41a51c75cf_>
> On Wednesday, December 16, 2015, Ben Parees <notifications@github.com>
> wrote: In test/extended/util/framework.go <#6329 (comment)
> <https://github.com/openshift/origin/pull/6329#discussion_r47846861>>: >
> @@ -189,7 +189,9 @@ var CheckImageStreamTagNotFoundFn = func(i
> *imageapi.ImageStream) bool { > func WaitForADeployment(client
> kclient.ReplicationControllerInterface, > name string, > isOK, isFailed
> func(*kapi.ReplicationController) bool) error { > - for { > + startTime :=
> time.Now() > + endTime := startTime.Add(3 * time.Minute) bump this up to
> something like 15 or 20 minutes. we don't want to introduce flakes if we
> have existing extended tests that take a long time waiting for a deployment
> for some reason. i'm also surprised 3 minutes is long enough for you test,
> given that it basically has to get through the entire imagebuild+deploy
> step in 3 minutes to pass. (3 minutes or the watch timeout, whichever is
> longer.) — Reply to this email directly or view it on GitHub <
> https://github.com/openshift/origin/pull/6329/files#r47846861>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6329/files#r47865373>.
>